sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -985,4 +976,59 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext
     val queryStats3 = query().queryExecution.optimizedPlan.stats.attributeStats
     assert(queryStats3.map(_._1.name).toSet === Set("c0", "v1", "v2"))
   }
+
+  test("Refresh Qualified Tables") {
+    withTempDatabase { db =>
+      withTempPath { path =>
+        spark.catalog.createTable(
+          s"$db.cachedTable",
+          "PARQUET",
+          StructType(Array(StructField("key", StringType))),
+          Map("LOCATION" -> path.toURI.toString))
+        withCache(s"$db.cachedTable") {
+          spark.catalog.cacheTable(s"$db.cachedTable", MEMORY_ONLY)
+          assertCached(spark.table(s"$db.cachedTable"), s"$db.cachedTable", MEMORY_ONLY)
+          assert(spark.catalog.isCached(s"$db.cachedTable"),
+            s"Table '$db.cachedTable' should be cached")
+
+          spark.catalog.refreshTable(s"$db.cachedTable")
+          assertCached(spark.table(s"$db.cachedTable"), s"$db.cachedTable", MEMORY_ONLY)
+          assert(spark.catalog.isCached(s"$db.cachedTable"),
+            s"Table '$db.cachedTable' should be cached after refresh")
+
+          activateDatabase(db) {
+            assertCached(spark.table("cachedTable"), s"$db.cachedTable", MEMORY_ONLY)
+            assert(spark.catalog.isCached("cachedTable"),
+              "Table 'cachedTable' should be cached after refresh")
+
+            spark.catalog.refreshTable(s"cachedTable")
+            assertCached(spark.table("cachedTable"), s"$db.cachedTable", MEMORY_ONLY)
+            assert(spark.catalog.isCached("cachedTable"),
+              "Table 'cachedTable' should be cached after refresh")
+          }
+        }
+      }
+    }
+  }
+
+  test("Refresh Unqualified Tables") {
```

Comment:
It's a little bit confusing with [the above test case with `activateDatabase`](https://github.com/apache/spark/pull/24221/files#diff-aabe595bd56d91bd109141f7452e39d1R1004). What do you mean `Refresh Unqualified Tables` here?

Could you elaborate the intention of this test case? Is that related with `sql("select * from cachedTable")`?

Reply:
Yes. They are confusing.. I refactor the test to have two sections. The first section caches a table with qualified table name and then refreshes the table with its unqualified table name. The second section caches a table with its unqualified table name and then refreshes the table with its qualified table name. They help to make sure cache name would not be updated due to table refreshing. I hope the test looks better now.